daemon/containerd/image_push.go

```diff
@@ -134,24 +134,22 @@ func (i *ImageService) pushRef(ctx context.Context, targetRef reference.Named, m
 		return err
 	}
 
-	addChildrenToJobs := containerdimages.HandlerFunc(
+	addLayerJobs := containerdimages.HandlerFunc(
 		func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
-			children, err := containerdimages.Children(ctx, store, desc)
-			if err != nil {
-				return nil, err
-			}
-			for _, c := range children {
-				jobsQueue.Add(c)
+			switch {
+			case containerdimages.IsIndexType(desc.MediaType),
+				containerdimages.IsManifestType(desc.MediaType),
+				containerdimages.IsConfigType(desc.MediaType):
```

Comment:
So, I'm considering if this is information we _would_ want to show at some point (after all, if we're pushing these, why would we not show progress?)

That said, maybe the current presentation of these isn't ideal, if they can be mistaken for layers. Wondering if there would be a way to fix the presentation, or have additional metadata so that the client can decide whether or not to show these (without discarding the actual progress information).


If we _do_ decide to discard this information, then at least it may be good putting a comment in this switch block, to explain why we're skipping these (possibly with either a tracking ticket or a link to the conversation on this PR)

Reply:
I think it would be great to keep them, but I think we should be consistent as we've already removed it from the pull progress for the same reason (https://github.com/moby/moby/pull/46412).

Technically we could use the `Progress.Aux` to pass additional information (like `ocispec.Descriptor`) so the client could decide (it would be visible by default for clients not aware of the extra information though).

Reply:
Ah, yes, I see I commented on that PR as well, and also looks like it was related to this 😅 so at least a code comment (perhaps also on the pull side) would make sense.

Do you recall if there were existing tests failing because of this? Slightly wondering what the consequences would be if we'd keep these (although the presentation part is a bit of an issue I guess)

Reply:
No, I don't see any test that's failing because of it.
So, maybe we can ignore the inconsistency with pull and keep them?